packages/flutter/lib/src/material/drop_down.dart

```diff
@@ -377,6 +414,21 @@ class _DropDownButtonState<T> extends State<DropDownButton<T>> {
       _updateSelectedIndex();
   }
 
+  @override
+  void dispose() {
+    if (_currentRoute != null) {
+      assert(_currentRoute.isActive);
+      Navigator.openTransaction(context, (NavigatorTransaction transaction) {
+        transaction.popUntil(_currentRoute);
+        assert(_currentRoute.isCurrent);
+        transaction.pop();
+      });
```

Comment:
Filed https://github.com/flutter/flutter/issues/4432 to track this. I've removed the code from this patch.